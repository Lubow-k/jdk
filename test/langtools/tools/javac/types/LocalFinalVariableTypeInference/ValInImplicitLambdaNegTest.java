/*
 * @test /nodynamiccopyright/
 * @summary compiler support for local-variable syntax for lambda parameters
 * @compile/fail/ref=ValInImplicitLambdaNegTest.out -XDrawDiagnostics ValInImplicitLambdaNegTest.java
 * @compile/fail/ref=ValInImplicitLambdaNegTest_source10.out -source 10 -XDrawDiagnostics ValInImplicitLambdaNegTest.java
 */

import java.util.function.*;

class ValInImplicitLambdaNegTest {
    IntBinaryOperator f1 = (x, val y) -> x + y;                              // error implicit and val
    IntBinaryOperator f2 = (val x, y) -> x + y;                              // error val and implicit
    IntBinaryOperator f3 = (int x, val y) -> x + y;                          // error val and explicit

    BiFunction<String[], String, String> f4 = (val s1[], val s2) -> s2;      // error val and array

    IntBinaryOperator f5 = (val x, val y) -> x + y;                          // ok
}
