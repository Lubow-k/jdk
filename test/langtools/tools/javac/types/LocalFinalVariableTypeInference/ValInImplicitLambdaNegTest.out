ValInImplicitLambdaNegTest.java:11:28: compiler.err.invalid.lambda.parameter.declaration: (compiler.misc.var.and.implicit.not.allowed)
ValInImplicitLambdaNegTest.java:12:28: compiler.err.invalid.lambda.parameter.declaration: (compiler.misc.var.and.implicit.not.allowed)
ValInImplicitLambdaNegTest.java:13:28: compiler.err.invalid.lambda.parameter.declaration: (compiler.misc.var.and.explicit.not.allowed)
ValInImplicitLambdaNegTest.java:15:52: compiler.err.restricted.type.not.allowed.array: val
4 errors